src/main/java/bio/terra/app/controller/SnapshotRequestApiController.java

```diff
@@ -0,0 +1,68 @@
+package bio.terra.app.controller;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.common.iam.AuthenticatedUserRequestFactory;
+import bio.terra.controller.SnapshotRequestApi;
+import bio.terra.model.EnumerateSnapshotAccessRequest;
+import bio.terra.model.SnapshotAccessRequest;
+import bio.terra.model.SnapshotAccessRequestResponse;
+import bio.terra.service.auth.iam.IamAction;
+import bio.terra.service.auth.iam.IamResourceType;
+import bio.terra.service.auth.iam.IamRole;
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.snapshotbuilder.SnapshotBuilderService;
+import io.swagger.annotations.Api;
+import jakarta.servlet.http.HttpServletRequest;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+@Controller
+@Api(tags = {"SnapshotRequest"})
+public class SnapshotRequestApiController implements SnapshotRequestApi {
```

Comment:
What should we call this? Options include:

1. SnapshotRequestApiController
2. SnapshotBuilderRequestApiController
3. SnapshotAccessRequestApiController
4. Other?

Reply:
I'm leaning toward `SnapshotAccessRequest`, for this method, the API endpoints, etc.

Reply:
I will change it for everything except the DAO, cause I would love the DAO to match the table. (And leave the Sam methods alone, since the resource type is still `snapshot_builder_request` - we could change that if we wanted but we should do it now if so.